src/sciline/task_graph.py

```diff
@@ -78,4 +76,11 @@ def visualize(
         """
         from .visualize import to_graphviz
 
-        return to_graphviz(self._graph, **kwargs)
+        class _GraphVizualization:
+            def __init__(self, graph):
+                self.graph = graph
+
+            def _repr_html_(self):
+                return self.graph.pipe(format='svg', encoding='utf8')
```

Comment:
Maybe this class should be moved into `visualize.py`? Or its own file, so we can use a proper type-hint without the `# type: ignore`?